src/components/PriorityBadge/PriorityBadge.tsx

```diff
@@ -0,0 +1,245 @@
+    /**
+     * Children nodes to be rendered within the component,
+     * specifically used to display the selected value from parent component.
+     */
+    children: React.ReactNode;
+    /**
+     * Optional object containing labels for different priorities, used primarily for localization.
+     */
+    priorityLabels?: {
+        mandatory?: string;
+        important?: string;
+        optional?: string;
+        exclude?: string;
+    };
+    /**
+     * Current priority value that indicates the importance of the component.
+     * Uses the Priority type which can be 'mandatory', 'important', 'optional', or 'exclude'.
+     */
+    priority: Priority;
+    /**
+     * Optional object specifying the currently selected option with value and label properties.
+     */
+    option?: Option;
+    /**
+     * Array of options available for selection.
+     * Each option is an object with a value and a label.
+     */
+    optionList: Array<Option>;
+    /**
+     * Optional header title for the options list when displayed, such as in a dropdown.
+     */
+    optionListHeader?: string;
+    /**
+     * Function to be called when the close button is clicked.
+     */
+    onClose?: (e: React.KeyboardEvent | React.MouseEvent) => void;
+    /**
+     * Callback function triggered when a new priority is selected.
+     * Provides the newly selected priority as an argument.
+     */
+    onPriorityChange?: (newPriority: Priority) => void;
+    /**
+     * Callback function triggered when a new option is selected.
+     * Provides the newly selected option object as an argument.
+     */
+    onChange: (newOption: Option) => void;
+    /**
+     * Boolean indicating whether the close button should be disabled.
+     * Can be used to prevent closing under certain conditions.
+     */
+    isCloseButtonDisabled?: boolean;
+    /**
+     * Boolean indicating whether the priority selection is disabled.
+     * Used to lock priority changes in specific situations.
+     */
+    isPriorityButtonDisabled?: boolean;
+    /**
+     * Boolean indicating whether the option selection is disabled.
+     * Useful for controlling access to option changes dynamically.
+     */
+    isOptionButtonDisabled?: boolean;
+}
+
+const { block, elem } = bem('PriorityBadge', styles);
+
+export const PriorityBadge: React.FC<Props> = ({
+    children,
+    isCloseButtonDisabled = false,
+    isOptionButtonDisabled = false,
+    isPriorityButtonDisabled = false,
+    onChange,
+    onClose = undefined,
+    onPriorityChange = undefined,
+    option,
+    optionList,
+    optionListHeader,
+    priority = 'mandatory',
+    priorityLabels,
+    ...rest
+}) => {
+    const [dropdownStates, setDropdownStates] = React.useState({
+        priority: false,
+        option: false,
+    });
+
+    const badgeRef = React.useRef<HTMLDivElement | null>(null);
+
+    const hasPriorityLabels = priorityLabels && Object.keys(priorityLabels).length > 0;
+
+    const iconMap = React.useMemo(
```

Comment:
I think we can declare `iconMap` outside of component and keep code simple, without `React.useMemo`.